Backpack/TabBarController/Classes/BPKTabBarItem.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2021 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class BPKTabBarDotImageDefinition;
+/**
+ * `BPKTabBarItem` is a subclass of `UITabBarItem` configured with Skyscanner style properties.
+ */
+IB_DESIGNABLE @interface BPKTabBarItem : UITabBarItem
+
+@property(nullable, strong) UIImage *originalImage;
+
+@property (nonatomic) UIUserInterfaceStyle interfaceStyle;
```

Comment:
The issue is that the `TabBarItem` doesn't have a `traitCollection` as it's just a sumb object that represents the UI. So we're reliant on this value being set by the `TabBarController`.

Reply:
Something will still happen if it's not set explicitly though(I don't remember if it will have a garbage value or if it'll be `0`). Do we know that we always set it before we try and use it?

Reply:
As things stand there could be a race condition, so we should definitely set it explicitly.

Reply:
Actually that's not true - it's an enum so it will default to the first value which is `UIUserInterfaceStyleUnspecified`